ee/clickhouse/models/property.py

```diff
@@ -266,6 +268,24 @@ def parse_prop_clauses(
                 final.append(f"{property_operator} {table_name}distinct_id IN ({subquery})")
             params.update(filter_params)
 
+        elif prop.type in ("performed_event", "performed_event_multiple"):
+            subquery, subquery_params = performed_event_subquery(prop=prop, team_id=team_id, prepend=prepend)
+            final.append(subquery)
+            params.update(subquery_params)
+
+        elif prop.type in ("stopped_performing_event", "performing_event_regularly", "restarted_performing event"):
+            # TODO: implement lifecycle query here
+            pass
+
+        elif prop.type in ("performed_event_sequence"):
+            # TODO: implement sequence query here
+            pass
+
+        elif prop.type in ("performed_event_first_time"):
```

Comment:
The idea here is that parse_prop_clauses can handle building the subqueries in a filter context. The underlying functions like `performed_event_subquery` should return the actual subquery to be used in a filter.

We should build a separate util/class that will intelligently build a standalone cohort query. `Parse_prop_clauses` builds filters with the EventQuery as a base query. In cohorts, we not necessarily filtering events but instead filtering for persons which changes how these filters are applied and interact